### PR TITLE
Copy menu item

### PR DESCRIPTION
**Pull request by WilliamTraoreee**
# What does this PR do?

Add the possibility to have a button to copy on the clipboard a string on a menu item

image

---

### Global

- [x] This PR does not introduce any breaking change
- [ ] This PR introduces breaking change(s) and has been labeled as such
- [x] I have found someone to review this PR and pinged him

### Store

- [ ] This PR introduces new store changes

### NX

- [x] I have run the dep-graph locally and made sure the tree was clean i.e no circular dependencies
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`

### Clean Code

- [x] I made sure the code is type safe (no any)
- [ ] I have included a feature flag on my feature, if applicable

### CODE REVIEW

File: libs/pages/application/ui/src/lib/components/container/container.tsx [hunk 1]
```diff
@@ -50,6 +50,8 @@ export function Container(props: ContainerProps) {
           url: link.url || '',
           external: true,
         },
+        copy: link.url || '',
```
Comment (bdebon) [inline libs/pages/application/ui/src/lib/components/container/container.tsx:53]: Why not accepting null for copy? In this case if copy is null you just don't show nothing

File: libs/shared/ui/src/lib/components/menu/menu-item/menu-item.tsx [hunk 2]
```diff
@@ -1,20 +1,44 @@
...
 import { Link, useNavigate } from 'react-router-dom'
+import ButtonIcon, { ButtonIconStyle } from '../../buttons/button-icon/button-icon'
+import Tooltip from '../../tooltip/tooltip'
 export interface MenuItemProps {
   name: string
   link?: { url: string; external?: boolean }
   contentLeft?: React.ReactNode
   contentRight?: React.ReactNode
   onClick?: (e: ClickEvent) => void
+  copy?: string
```
Comment (bdebon) [inline libs/shared/ui/src/lib/components/menu/menu-item/menu-item.tsx:12]: Ah here you seems to accept that copy is null

**Review by RemiBonnet — CHANGES_REQUESTED** [review]
I am agree with the request by Benjamin, and can you hide the button when we don't have link? example here : `/organization/ffffffff-ffff-ffff-ffff-ffffffffffff/project/e0aabc0d-99cb-4867-ad39-332d6162c32c/environment/b91d2eb8-a850-49b5-8626-ade7afc4a28b/application/abbcf976-27a1-4531-9cdd-e4d15d7b2c27/general` 

Capture d’écran 2022-05-30 à 15 04 18

Plus, the height of this button has change, can you fix that?

Thanks!

File: libs/pages/application/ui/src/lib/components/container/container.tsx [hunk 3]
```diff
@@ -42,6 +42,8 @@ export function Container(props: ContainerProps) {
 
   const menuLink: MenuData = []
 
+  console.log(application)
```
Comment (RemiBonnet) [inline libs/pages/application/ui/src/lib/components/container/container.tsx:45]: need to be removed

File: libs/pages/application/ui/src/lib/components/container/container.tsx [hunk 4]
```diff
@@ -74,16 +75,27 @@ export function Container(props: ContainerProps) {
...
+      )
+    }
+  }
+
+  const headerButtons = (
+    <div>
+      {/*<ButtonIcon icon="icon-solid-terminal" style={ButtonIconStyle.STROKED} />*/}
+      {/*<ButtonIcon icon="icon-solid-scroll" style={ButtonIconStyle.STROKED} />*/}
+      {/*<ButtonIcon icon="icon-solid-clock-rotate-left" style={ButtonIconStyle.STROKED} />*/}
+      {application?.links && application.links.items && application.links.items.length !== 0 && linkButtons()}
```
Comment (RemiBonnet) [inline libs/pages/application/ui/src/lib/components/container/container.tsx:98]: `application.links.items.length > 0` 

cleaner to have this

Comment (RemiBonnet) [inline libs/pages/application/ui/src/lib/components/container/container.tsx:98]: `application.links.items.length > 0` 

cleaner to have this

**Comment by RemiBonnet** [comment]
@WilliamTraoreee ok for merge!

**Pull request merged**